docs: document the cypress tap CLI (#6785)
cypress-io/cypress-documentation: docs/cloud/features/cypress-ai-features.mdx

| Understand Tests  | `cypress-explain` Skill                    | Helps AI describe, and critique existing tests.                                                                  |
| Analyze & Triage  | Cloud MCP                                  | Gives AI coding assistants and AI agents a direct window into your application's health and stability            |
| Triage & Automate | Cloud CLI                                  | Brings Cloud data and Test Replay access to the terminal for triage, scripting, CI, and terminal-based AI agents |

> | Run & Debug Tests | `cypress tap` CLI                          | Lets AI agents drive a live Cypress session: run specs, read failures, and inspect the app under test            |

Reviewer: ```suggestion
| Run & Debug Tests | `cypress tap` CLI                          | Live Cypress app access for your agent. Drives Cypress and pulls the command log and DOM directly into context.          |
```

Read the [Cloud CLI documentation →](/cloud/integrations/cloud-cli)

### cypress tap

:::note

> **Best for:** AI coding agents that verify their own changes against a live Cypress session while they code.

Reviewer: ```suggestion
**Best for:** AI coding agents that need live app context, DOM, command logs, to check their work against a running Cypress session.
```

:::

`cypress tap` is an extension of the `cypress` CLI that enables driving and inspecting an open-mode Cypress
session from the terminal. An agent can run a spec and poll its status, then

> read the failing test's Command Log and error the way the reporter shows them.

Reviewer: ```suggestion
read the failing test's Command Log and error as the reporter shows them.
```

It can also inspect the DOM of the application-under-test at the moment a command ran.

Where the Cloud CLI and Cloud MCP give agents your recorded run data in

> Cypress Cloud, `cypress tap` gives your AI agent the direct access to your Cypress session on your

Reviewer: ```suggestion
Cypress Cloud, `cypress tap` gives your AI agent context to your Cypress session and your application-under-test on your
```

Author: Very fair!

> machine, so failures are found and fixed before the code ever reaches CI.

Reviewer: ```suggestion
machine, while developing and verifying tests before it ever runs in CI.
```

Read the [Cloud CLI documentation →](/cloud/integrations/cloud-cli)

### cypress tap

:::note
**Best for:** AI coding agents that verify their own changes against a live Cypress session while they code.
:::

> `cypress tap` is a set of CLI commands that drive a running `cypress open`

Reviewer: ```suggestion
`cypress tap` is an extension of the `cypress` CLI that enables driving and inspecting an open-mode Cypress
```

session from the terminal. An agent can run a spec and poll its status, then
read the failing test's Command Log and error the way the reporter shows them.

> It can also inspect the DOM of the app under test at the moment a command ran.

Reviewer: ```suggestion
It can also inspect the DOM of the application-under-test at the moment a command ran.
```

> Every command outputs JSON built for automation.

Reviewer: ```suggestion
```

Where the Cloud CLI and Cloud MCP give agents your recorded run data in

> Cypress Cloud, `cypress tap` gives your AI agent the live session on your

Reviewer: ```suggestion
Cypress Cloud, `cypress tap` gives your AI agent the direct access to your Cypress session on your
```

Reviewer: Agents can use cypress tap to pull Cypress state, DOM, etc to add context to decide what the correct next command to author is or how to fix an issue in their test

- [Cloud MCP](/cloud/integrations/cloud-mcp)
- [Cloud CLI](/cloud/integrations/cloud-cli)
- [AI Skills](/app/tooling/ai-skills)

> - [cypress tap: drive a live Cypress session from the terminal](/app/tooling/cypress-tap)

Reviewer: ```suggestion
- [cypress tap: live Cypress app access for your agent](/app/tooling/cypress-tap)
```
